src/seedu/addressbook/Main.java

```diff
@@ -106,11 +106,14 @@ private void recordResult(CommandResult result) {
      * @return result of the command
      */
     private CommandResult executeCommand(Command command)  {
+        command.setData(addressBook, lastShownList);
+        CommandResult result = command.execute();
         try {
-            command.setData(addressBook, lastShownList);
-            CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
+        } catch (StorageOperationException e) {
+            ui.showToUser(e.getMessage());
```

Comment:
This was from a previous LO? 
Do note that new branches should be made from the master branch, to prevent changes from other branches entering it.